fili-core/src/main/java/com/yahoo/bard/webservice/application/DruidDimensionsLoader.java

```diff
@@ -84,6 +84,54 @@ public DruidDimensionsLoader(
             DimensionDictionary dimensionDictionary,
             DruidWebService druidWebService
     ) {
+        this(
+                druidWebService,
+                buildDimensionList(dimensionDictionary),
+                buildDataSourcesList(physicalTableDictionary)
+        );
+    }
+
+    /**
+     * Builds a list of datasources from the physical tables.
+     * @param physicalTableDictionary  The physical tables to load from.
+     * @return the list of datasources.
+     */
+    public static List<DataSource> buildDataSourcesList(PhysicalTableDictionary physicalTableDictionary) {
```

Comment:
👍 It's probably better to have another constructor that takes the dimensions and the physical table dictionary and converts the dict into the datasources, instead of having the convertToDatasources method getting called as part of the call to some other constructor outside this class.